### DISCUSSION TITLE
Loss of theme and downgrading help

### DISCUSSION THREAD

**Discussion by bulkiapp** [florisboard/florisboard#1796] [Q&A] [answered]
I have been using Florisboard 0.3.13 for a few months now, and today I made a mistake by upgrading to 0.3.15(80). I am using Android 12 with a Samsung Note 10+ for the record.

Seeing how there is not a simple way available to customize the theme and adjust the spacing of the keys (or recover my old config in a straightforward way), I would like to ask where the directory for those configs is, so I can back up my old custom theme before i downgrade. I've tried going through the phone with a file manager, but I can't find the dev.patrickgold.florisboard folder at all.

I was curious about this because I am afraid uninstalling the current version and replacing it with the previous one I was using will wipe the parametres and any user customizations. 

I found 0.3.13 much simpler, cleaner and better to use overall, the quick bar on top in the new version is quite annoying with the clipboard that can't be removed, I can't copy and paste straight from the keyboard anymore with just 2 taps, it's really cluttered and unintuitive on .15

I'm sure things will get ironed out overtime, but for the time being, I'd like to get my old config back and stick with it. Could you guys help me out with that?

**Comment by patrickgold** [comment, marked as answer]
That's sad to hear that you do not like the new look and feel of FlorisBoard. I've read through your text and there are some points that I want to mention trying to convince you to stay on the new version.

First, you can configure the key spacing as previously by going to Settings > Keyboard > Key spacing. There, you can set the desired spacing for vertical and horizontal. Same goes for keyboard height, is just below that preference.

Secondly, you can switch the Smartbar primary and secondary layout (Settings > Smartbar > Row layout), so the clipboard cursor tools are displayed primarily and the secondary row has the undo/redo... stuff. This way the clipboard cursor tools are primarily displayed.

Regarding the themes and generally losing the config on upgrade: I am sorry that I wasn't able to get in a migration tool, but it would have taken me quite a bit of time and time is the thing that I have least, so to keep the project progression on track I wasn't able to do it. (Also themes have structurally changed so much internally that automatic migration would have been really difficult).

As for the theme creation, I agree that getting to the stylesheet editor got more complicated (and I need to work on that to make it easier), but the actual stylesheet editor is now way better than the 0.3.13 one, based on a lot of feedback I received in the 0.3.14 dev cycle. To create a new theme now: Settings > Theme > Manage installed themes > Create theme extension > Click on the orange plus > Select which base theme you want > Create. Then Edit, in there you can edit the colors and also get a live preview by tapping into the preview field. When finished Save the theme, then select it either for day or night. Like I said and also feedback I received the creation process has too many steps, but once it is created the actual theme design is easier. I will work on this and make it simpler in the following releases.

If I haven't yet convinced you to stay on 0.3.15 I can try to help you downgrade, but without a rooted device this is near impossible to do so without losing data (especially since proper backup was only introduced in 0.3.14).

The files are located in `/data/user/0/dev.patrickgold.florisboard`. This folder is not accessible for "normal" users and also not visible in your file explorer, simply because it can hold private files that is app-specific in scope. You could theoretically try to force an adb backup, but I don't know if it was doable on the 0.3.13 APK (the app-manifest allows this only since 0.3.14). Else I currently have no idea how you could retrieve the data, I am sorry.

I hope that this answer helps, else don't hesitate to ask for more assistance!

**Reply by bulkiapp** [reply]
I really appreciate the quick reply and thorough explanation of the features of 0.3.15. It might have felt more overwhelming than it is really, and I know you are working really hard on it so I want to thank you again for the time spent developing florisboard and replying to my question.

As soon as I find some time I will try to replicate the setup I had on 0.3.13 on the newest update, since it was pretty simple. This is close to what it looked like (had some height and width slight adjustments but nothing too crazy)

![signal-2022-04-18-175614_001](https://user-images.githubusercontent.com/103817820/163835710-6fcd0e38-65e5-4705-a952-60edae74d7de.jpeg)

I sent you 3.50€ through your Paypal.me link since you deserve it for all the effort put into this great open source keyboard! 
Thanks again!
